### ISSUE TITLE
drun mode executes *.desktop files with wrong priority

### ISSUE THREAD

**Issue by cryzed** [DaveDavenport/rofi#719]
## Version

Version: 1.4.2

## Configuration

https://gist.github.com/cryzed/ee80d3852b82789c4b6c19b0ff8547ac

## Launch Command

`$ rofi -show combi -combi-modi "window,drun,run" -modi combi`

## Steps to reproduce

* Have a *.desktop file in `~/.local/share/applications` that overrides a *.desktop file of the same name in `/usr/share/applications` (e.g. `Discord.desktop`)
* Launch the desktop file using rofi's drun mode

## What behaviour you see

* Launching the desktop file using rofi's drun mode executes `/usr/share/application/Discord.desktop` instead of `~/.local/share/applications/Discord.desktop`.

## What behaviour you expect to see

Launching the desktop file using rofi's drun mode executes the appropriate command in respect to the desktop files.

This is a problem when using custom *.desktop files to execute programs in a firejail environment for example.

**Comment by sardemff7** [comment]
Could you share the rofi output with `export G_MESSAGES_DEBUG=Dialogs.DRun` and both `.desktop` files, please?

**Comment by cryzed** [comment]
- [~/.local/share/applications/Discord.desktop](https://gist.github.com/cryzed/85e8469bbafdb9e0ddefe988e2b45d32)
- [/usr/share/application/Discord.desktop](https://gist.github.com/cryzed/7dc201f5c9db2cbe70e3bcb48b6f2a86)
- [rofi output](https://gist.githubusercontent.com/cryzed/c31cf1a422da1876d772063414d338fe/raw/f39b957365d13f16160894fa0c62b441127027f9/gistfile1.txt)

The important difference is here that the path in the `Exec` command is _relative_ in the local *.desktop file, which on my end should prefer the firejail wrapper script located in `/usr/local/bin`.

**Comment by sardemff7** [comment]
From the log, you can see:
```
(process:12134): Dialogs.DRun-DEBUG: Skipping: Discord.desktop, was previously seen.
(process:12134): Dialogs.DRun-DEBUG: Skipping: Discord.desktop, was previously seen.
```
The first is while walking `~/.local/share/applications`, second is for `/usr/share/applications`. That means you have a third file, in a subdir of `~/.local/share/applications`, that is read first.

Then, the file you pasted has `Exec=discord`, this is not relative to the `.desktop` file, it means “search in `$PATH`”, but many that’s what you meant with “relative”? (There have been a lot of discussions about `.desktop` files supporting relative `Exec=` key to support sandboxing (flatpack and friends) so we must be clear here.)

**Comment by cryzed** [comment]
I don't have a third Discord *.desktop file:
```user@host ~ $ find /usr/share/applications/ ~/.local/share/applications/ -iname '*discord*'
/usr/share/applications/Discord.desktop
/home/user/.local/share/applications/Discord.desktop
```

And yes, I meant "search in `$PATH`", yes:
```[user@host ~]$ echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/user/.bin
```

**Comment by sardemff7** [comment]
Could you please build rofi `next` branch and paste a log with that build?

**Comment by cryzed** [comment]
That seems to have [worked](https://gist.github.com/cryzed/b2f16b26adae1bb944d394651c97a22c), Discord was launched properly using the firejail wrapper script.

**Comment by sardemff7** [comment]
Then it was probably the bug #579 fixed in 7929818be2e3b16437c37f50fc3fc7be5da4f658, related to history. I forgot about this one.
